pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,134 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    /**
+     * Options passed to the HTTP client constructor.
+     */
+    export interface HttpOptions {
+        /**
+         * The target address. If omitted, the endpoint string must include the host.
+         */
+        address?: string;
+        /**
+         * The port to connect to.
+         */
+        port?: number;
+        /**
+         * Enable TLS (HTTPS) connection.
+         */
+        tls?: boolean;
+        /**
+         * Whether to connect as superuser.
+         */
+        superuser?: boolean;
+        /**
+         * If true, responses will be delivered as binary (Uint8Array) instead of string.
+         */
+        binary?: boolean;
+        /**
+         * Default HTTP headers to send with every request.
+         */
+        headers?: { [key: string]: string };
+        /**
+         * Default query parameters to include with every request.
+         */
+        params?: { [key: string]: string | number };
+    }
+
+    /**
+     * Allowed types for HTTP request bodies.
+     * A body may be a string or a plain JavaScript object that will be encoded as JSON.
+     * If the body is undefined or null, an empty HTTP body is sent.
+     */
+    export type HttpBody = string | { [key: string]: any } | null | undefined;
```

Comment:
Typechecker [still complains](https://github.com/cockpit-project/cockpit/actions/runs/13742924534/job/38468520857?pr=21678#step:2:471) about this `any` here. I think what you want here is `string | JsonObject | null`. The `undefined` should already be captured by the `?` optionality, but I figure it can't hurt to keep it.

But also see below.